resources/fetching/fetchers/azure/security_fetcher.go

```diff
@@ -0,0 +1,91 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package fetchers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/fetching/cycle"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib/inventory"
+)
+
+type AzureSecurityAssetFetcher struct {
+	log        *logp.Logger
+	resourceCh chan fetching.ResourceInfo
+	provider   azurelib.ProviderAPI
+}
+
+func NewAzureSecurityAssetFetcher(log *logp.Logger, ch chan fetching.ResourceInfo, provider azurelib.ProviderAPI) *AzureSecurityAssetFetcher {
+	return &AzureSecurityAssetFetcher{
+		log:        log,
+		resourceCh: ch,
+		provider:   provider,
+	}
+}
+
+func (f *AzureSecurityAssetFetcher) Fetch(ctx context.Context, cycleMetadata cycle.Metadata) error {
+	f.log.Info("Starting AzureSecurityAssetFetcher.Fetch")
+
+	subscriptions, err := f.provider.GetSubscriptions(ctx, cycleMetadata)
+	if err != nil {
+		return fmt.Errorf("error fetching subscription information: %w", err)
+	}
+
+	var errs []error
+
+	for _, sub := range subscriptions {
+		securityContacts, err := f.provider.ListSecurityContacts(ctx, sub.ShortID)
+		if err != nil {
+			f.log.Errorf("AzureSecurityAssetFetcher.Fetch failed to fetch contacts for subscription %s: %s", sub.ShortID, err.Error())
+			errs = append(errs, err)
+			continue
+		}
+		if securityContacts == nil {
```

Comment:
why check if it's null? As far as I understand the behaviour of empty slice and nil is the same (or very close to each other)

Reply:
I just wasn't sure what is better to store in elastic, empty array `[]` vs `null`, and chose the empty slice/array.